### ISSUE TITLE
ParamConverters are not supported

### ISSUE THREAD

**Issue by who** [kongchen/swagger-maven-plugin#141]
It looks like ParamConverters are not resolving correctly, and when they are used, an invalid spec is rendered by swagger-maven-plugin.

I have created a project to illustrate this here: 
https://github.com/who/jersey-param-converters/tree/examples/swagger-maven-plugin

This method ...

```
    @GET
    @Path("query")
    @ApiOperation(value = "getViaQueryParam")
    public Entity getViaQueryParam(
            @QueryParam("entity")
            final Entity entity) {
        return entity;
    }
```

... renders this spec ...

```
...
    "/json-param-converter/query" : {
      "get" : {
        "tags" : [ "JsonParamConverterResource" ],
        "summary" : "getViaQueryParam",
        "description" : "",
        "operationId" : "getViaQueryParam",
        "produces" : [ "application/json" ],
        "parameters" : [ {
          "name" : "entity",
          "in" : "query",
          "required" : false,
          "type" : "ref"
        } ],
        "responses" : {
          "200" : {
            "description" : "successful operation",
            "schema" : {
              "$ref" : "#/definitions/Entity"
            }
          }
        }
      }
    }
...
```

Note the above `"type" : "ref"` is not valid Swagger 2.0.

Here's a good intro to ParamConverters: http://blog.dejavu.sk/2014/02/11/inject-custom-java-types-via-jax-rs-parameter-annotations/

**Comment by kongchen** [comment]
[Swagger Sepc](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#parameter-object) do not support putting an object in query:
> Since the parameter is not located at the request body, it is limited to simple types (that is, not an object). The value MUST be one of "string", "number", "integer", "boolean", "array" or "file"

Yes, the type `ref` in incorrect here, however, it is hard to determine for this case, my thoughts is raise this issue in swagger's mail list.

**Comment by who** [comment]
You are correct, swagger spec does not support putting an object in a query. 

However, that's not entirely what's going on here.  The ParamConverters aren't the actual object being received in the query/path. For example, the path `/pets/{petName}` could accept a Pet class in it's controller method, but that's not what is actually being accepted by the endpoint; String is being accepted, and the ParamConverter automatically converts String `petName` to an object.

A workaround for this is to use ApiImplicitParams, to allow the manual definition of the param, but I don't know if swagger-maven-plugin supports that or not.

**Comment by kongchen** [comment]
Yes, I think we're on the same page.

To follow swagger spec now, there's only one option for the type -- string:
```
"parameters" : [ {
          "name" : "entity",
          "in" : "query",
          "required" : false,
          "type" : "string"
        } ]
```
But there should be somewhere to show what the string should be for the people who reads the document.

The only approach I can found now is to use `@ApiParam` and describe how to write the `string` in `description` field.

`@ApiImplicitParams` cannot be used because it will generate another parameter,  duplicate with the one you defined in `@QueryParam`, looks weird.

Anyway, it's a bug, the type of a parameter should not be `ref`, I'll fix it soon.

**Comment by who** [comment]
Thanks, confirmed fix in 3.0.1-SNAPSHOT
